prow.go

```diff
@@ -132,9 +132,9 @@ func (r *URLConfigResolver) Resolve(org, repo, branch, variant string) ([]byte,
 
 // stopJob triggers namespace deletion, which will cause graceful shutdown of the
 // cluster. If this method returns nil, it is safe to consider the cluster released.
-func (m *jobManager) stopJob(name string) error {
-	namespace := fmt.Sprintf("ci-ln-%s", namespaceSafeHash(name))
-	if err := m.projectClient.ProjectV1().Projects().Delete(context.TODO(), namespace, metav1.DeleteOptions{}); err != nil && !errors.IsNotFound(err) {
+func (m *jobManager) stopJob(job *JobRequest) error {
+	namespace := fmt.Sprintf("ci-ln-%s", namespaceSafeHash(job.Name))
+	if err := m.buildClusterClientConfigMap[job.BuildCluster].ProjectClient.ProjectV1().Projects().Delete(context.TODO(), namespace, metav1.DeleteOptions{}); err != nil && !errors.IsNotFound(err) {
```

Comment:
job is not threadsafe, you must put these behind threadsafe accessors.

Reply:
I added a mutex to the JobRequest.  It looks like all the other calls appear to be protected by a mutex at a higher level.

Reply:
Scratch my last comment.  I've moved the call to `stopJob` to within the JobManager mutex...
https://github.com/openshift/ci-chat-bot/blob/1301dbfd3d590318012fbe12ab1de9cb2b364c2e/manager.go#L1186-L1187